src/js/hooks/inboundV2/addItems/useInboundAddItemsColumns.jsx

```diff
@@ -0,0 +1,374 @@
+      ),
+      cell: ({ row }) => {
+        const hasErrors = !!errors?.[row.index]?.quantityRequested?.message;
+        return (
+          <TableCell
+            className="rt-td"
+            tooltip={hasErrors}
+            tooltipForm
+            tooltipLabel={hasErrors && errors[row.index].quantityRequested.message}
+          >
+            <Controller
+              name={`values.lineItems.${row.index}.quantityRequested`}
+              control={control}
+              render={({ field }) => (
+                <TextInput
+                  {...field}
+                  type="number"
+                  hasErrors={hasErrors}
+                  showErrorBorder={hasErrors}
+                  onBlur={() => {
+                    field.onBlur();
+                    trigger(`values.lineItems.${row.index}.quantityRequested`);
+                  }}
+                />
+              )}
+            />
+          </TableCell>
+        );
+      },
+      meta: {
+        flexWidth: 1,
+      },
+    }),
+    columnHelper.accessor('recipient', {
+      header: () => (
+        <TableHeaderCell style={{ justifyContent: 'center' }}>
+          {translate('react.stockMovement.recipient.label', 'Recipient')}
+        </TableHeaderCell>
+      ),
+      cell: ({ row }) => {
+        const hasErrors = !!errors?.[row.index]?.recipient?.message;
+        return (
+          <TableCell
+            className="rt-td"
+            tooltip={hasErrors}
+            tooltipForm
+            tooltipLabel={hasErrors && errors[row.index].recipient.message}
+          >
+            <Controller
+              name={`values.lineItems.${row.index}.recipient`}
+              control={control}
+              render={({ field }) => (
+                <SelectField
+                  {...field}
+                  options={users}
+                  hasErrors={hasErrors}
+                  placeholder={getCustomSelectErrorPlaceholder({
+                    id: 'react.stockMovement.recipient.label',
+                    defaultMessage: 'Recipient',
+                    displayIcon: hasErrors,
+                  })}
+                />
+              )}
+            />
+          </TableCell>
+        );
+      },
+      meta: {
+        flexWidth: 1.5,
+      },
+    }),
+    columnHelper.display({
+      id: 'delete',
+      header: () => (
+        <TableHeaderCell style={{ justifyContent: 'center' }}>
+          {translate('react.default.button.delete.label', 'Delete')}
+        </TableHeaderCell>
+      ),
+      cell: ({ row }) => (
+        <TableCell className="rt-td">
+          <div className="bin-container">
+            <RiDeleteBinLine
+              className="inbound-bin"
+              display={row?.original?.statusCode === 'SUBSTITUTED'}
+              onClick={() => handleDelete(row)}
+            />
+          </div>
+        </TableCell>
+      ),
+      meta: {
+        flexWidth: 0.5,
+      },
+    }),
+  ], [errors, control]);
```

Comment:
errors are presumably list or object. We want dependency to be a simple value like string, number or a boolean. Otherwise this would be recreated over and over again.
I think some dependencies are still missing, like language.